mailbox/elasticsearch/src/test/java/org/apache/james/mailbox/elasticsearch/ElasticSearchIntegrationTest.java

```diff
@@ -104,4 +122,73 @@ protected void initializeMailboxManager() throws Exception {
         storeMailboxManager.setMessageSearchIndex(messageSearchIndex);
         storeMailboxManager.init();
     }
+
+    @Ignore("MAILBOX-301 Lucene terms length exceeded on some emails")
+    @Test
+    public void termsBetweenElasticSearchAndLuceneLimitDueTuNonAsciiCharsShouldNotMakeIndexingFail() throws Exception {
```

Comment:
https://www.elastic.co/guide/en/elasticsearch/guide/current/multi-fields.html

Read this and you will understand the **_source** is stored only once. In the **textBody.raw**. Which ignores terms of more than 4KB. Hence, my fix works!

(You don't trust tests and TDD?)